File: foilingTrimaran_ThomasSauder/weightLoad.m
function weightLoadInB = weightLoad(eta,verbose)
% Returns the load expressed in {b}

% Mass propoerties
boatMass = 14.4e3   ;
COGpositionInB = [-1.3; 0; -2.4];
g=9.81;


weightForceInB = Rbn(eta).'*[0;0;boatMass*g];
weightLoadInB = [weightForceInB;cross(COGpositionInB,weightForceInB)];
% if verbose
%     fprintf(1,'w  |                                                | %+10.1f %+10.1f %+10.1f | %+10.1f %+10.1f %+10.1f | \n',weightLoadInB.'/1000);
% end
end

File: rlController/weightLoad.py
import numpy as np
from Rbn import Rbn

def weightLoad(eta, verbose=False):
    """
    Returns the weight load expressed in {b}
    """

    # --------------------------------------------------
    # Mass properties
    # --------------------------------------------------
    boatMass = 2.3e3 + 87.5 * 6   # 2,332–2,432 kg
    COGpositionInB = np.array([5.92896, 0.07904, -3.86752])
    g = 9.81

    # --------------------------------------------------
    # Weight force in {b}
    # --------------------------------------------------
    weightForceInB = Rbn(eta).T @ np.array([0.0, 0.0, boatMass * g])

    weightLoadInB = np.hstack((
        weightForceInB,
        np.cross(COGpositionInB, weightForceInB)
    ))

    return weightLoadInB
